### PR TITLE
Add support for MK3S Revo HT 40W hot end

### PR DESCRIPTION
**Pull request by b**
This code adds support to the MK3S for the e3d Revo high temperature hot end with the 40W precision heater and PT1000 thermistor. Works on my machine.

### CODE REVIEW

**Comment by 3d-gussner** [comment]
@b Thanks for the PR I forwarded this to E3D as your PR ONLY contains ONE set of data points.

File: Firmware/variants/MK3S-E3DREVO_HT_40W.h [hunk 1]
```diff
@@ -0,0 +1,674 @@
...
+#define PRINTER_TYPE PRINTER_MK3S
+#define PRINTER_NAME PRINTER_MK3S_NAME
+#define PRINTER_NAME_ALTERNATE PRINTER_MK3_NAME //the other similar printer to this.
+#define PRINTER_MMU_TYPE PRINTER_MK3S_MMU3
+#define PRINTER_MMU_NAME PRINTER_MK3S_MMU3_NAME
+#define FILAMENT_SIZE "1_75mm_MK3S"
+#define NOZZLE_TYPE "E3DREVO_HT_40W"
+
+// Printer name
+#define CUSTOM_MENDEL_NAME "Prusa i3 MK3S-RHT40"
```
Comment (3d-gussner) [inline Firmware/variants/MK3S-E3DREVO_HT_40W.h]: Name is to long it is limited to 17 chars at the moment. Also the PR contains a PT1000 thermistor and as there is a HT PT100 version available the naming should also be different.
```
01234567890123456789
Prusa MK3S-RHT1k          //for E3d Revo HT with PT1000
```
the other 3 options could be

``` 
01234567890123456789
Prusa MK3S-RHT1c          //for E3d Revo HT with PT100 
Prusa MK3S-RUHT1c       //for E3d Revo UHT with PT100
Prusa MK3S-RUHT1k       //for E3d Revo UHT with PT1000
```

**Comment by b** [comment]
Is there anything else I need to do to make this PR acceptable for merging?

**Comment by 3d-gussner** [comment]
@b I have been in contact with E3D about this PR and there are few reasons we didn't approved this PR
1. The max high temperature of 405°C may soften the plastic parts which are made in PETG (ASA/ABS fan-duct). The extruder would need to be printed in a different material that can withstand the heat radiation.
2. The TM values base only on YOUR printer and will not represent the wider possible range.
  a. For the E3DRevo TM values we collected multiple test results from the community and E3D also provided more data to get  "good" average TM values for these.
  b.  So you may need to ask other community members to test and update then the TM values
3. Each variant must have it's own variant and TM file. Please don't use [this](https://github.com/prusa3d/Prusa-Firmware/pull/4487/files#diff-42e00f9f4ef21f29b672e717d34445ef7950285670d4d3dde69c0586658ec86aR39-R42) and [this](https://github.com/prusa3d/Prusa-Firmware/pull/4487/files#diff-42e00f9f4ef21f29b672e717d34445ef7950285670d4d3dde69c0586658ec86aR39-R42) anymore
  a. I know there is the existing `#elif defined` but this is an very old community merge and we won't change it

Hope that you get some help from the community so we can consider to review this PR again.
The provided the E3D REVO hex files in the official release builds are an exception as E3D approved these.
But please understand that we will not build and provide the hex files for each 3rd party variant there may come in the official releases. Meaning the community will have to compile the firmware themself.

**Comment by 3d-gussner** [comment]
Closing as no interaction or update

**Pull request closed without merging**